### ISSUE TITLE
Cann't build for armv7 with "neon"

### ISSUE THREAD

**Issue by Dirreke** [ziglang/zig#16018]
### Zig Version

0.9.1

### Steps to Reproduce and Observed Behavior

I use `zig c++ -target arm-linux-gnueabihf -mcpu=generic+v7a+vfp3+d32+thumb2+neon -mfpu=neon -mfloat-abi=hard`
and it still cann't add neon to target-feature, and I always get this
```
/opt/zig/zig-linux-x86_64-0.9.1/lib/include/arm_neon.h:32:2: error: "NEON support not enabled"
#error "NEON support not enabled"
```
Is it a bug? Or maybe I just missed some args?

### Expected Behavior

//

**Comment by leecannon** [comment]
do you have the same issue with the latest stable zig release 0.10.1?

**Comment by Dirreke** [comment]
I cann't build for arm with `0.10.1` because #6573 
```
  cargo:warning=warning(compilation): libc++ does not work on multi-threaded ARM yet.
  cargo:warning=For more details: https://github.com/ziglang/zig/issues/6573
  cargo:warning=error: unable to create compilation: TargetRequiresSingleThreaded
  exit status: 1
```

**Comment by iacore** [comment]
> is there any way to build like [`rocksdb`](https://github.com/facebook/rocksdb) or [`Cbc`](https://github.com/coin-or/Cbc) with `zig 0.10.1`?

Can you use Clang?

**Comment by andrewrk** [comment]
Zig is indeed passing the neon flags to Clang, so the question is why doesn't clang define the macro?

```c
#if !defined(__ARM_NEON)
#error "NEON support not enabled"
#else
```

Can you file a bug report on Clang? Use `ZIG_VERBOSE_CC=1` to observe the clang command that zig spits out (which includes `-Xclang +neon`) and ask them why `__ARM_NEON` is not defined.

**Comment by Dirreke** [comment]
I'd love to try it. However, I'm a little busy  these days. I will try to find this bug some days later.

**Comment by Dirreke** [comment]
It issue seems very related to #10411

**Comment by ldube** [comment]
(Tested on zig version 0.14.0-dev.1511+54b668f8a)

64bit ARMv8 defines __ARM_NEON and 32bit ARMv7 defines __ARM_FEATURE_SIMD32.

I used the commands below.

```
zig c++ -target aarch64-linux-gnueabihf  -x  c /dev/null -dM -E | grep -e SIMD -e NEON
#define __ARM_NEON 1
#define __ARM_NEON_FP 0xE

zig c++ -target arm-linux-gnueabihf  -x  c /dev/null -dM -E | grep -e SIMD -e NEON
#define __ARM_FEATURE_SIMD32 1

```

**Comment by alexrp** [comment]
It seems that `arm_neon.h` now checks for `__ARM_FP`. Is it usable now?

**Comment by ldube** [comment]
No. ```__ARM_NEON``` is checked because ```__ARM_FP``` is defined. 

To make the program at the end compile, the check would have to be changed to this:
```#if !defined(__ARM_NEON) && !defined(__ARM_FEATURE_SIMD32)```

```zig c++ -target arm-linux-gnueabihf -c hello.cpp```
```

#include <arm_neon.h>

poly8x8_t atest(void) {
  poly8x8_t a;
  return splat_lane_p8(a, 0);
}
```

**Comment by alexrp** [comment]
> `__ARM_NEON` is checked

Where?

> To make the program at the end compile

It compiles for me with those flags.

**Comment by ldube** [comment]
> > `__ARM_NEON` is checked
> 
> Where?
In arm_neon.h
> 
> > To make the program at the end compile
> 
> It compiles for me with those flags.
I probably have an outdated build. I'll update and try again.

**Comment by ldube** [comment]
I probably have an outdated build. I'll update and try again.

**Comment by alexrp** [comment]
`0.14.0-dev.1659+4ceefca14` here FWIW.

**Comment by ldube** [comment]
Works with ``` 0.14.0-dev.1651+ffd071f55 ```
